File: Builder/Dart/Report.dart
class Report {
  String? _filename;
  String? _content;

  void setFilename(String filename) {
    _filename = filename;
  }

  void setContent(String content) {
    _content = content;
  }

  void describe() {
    if (_filename != null) {
      print("The name of the file is $_filename");
    } else {
      print("File doesn't have any name");
    }

    if (_content != null) {
      print("The content of the file is $_content");
    } else {
      print("File doesn't have any content");
    }
  }
}

File: Builder/PHP/Report.php
<?php

class Report {
    private string $filename;
    private string $content;

    public function setFilename(string $filename) {
        $this->filename = $filename;
    }

    public function setContent(string $content) {
        $this->content = $content;
    }

    public function describe() {
        if ($this->filename) {
            echo "The name of the file is " . $this->filename . "\n";
        } else {
            echo "The file doesn't have any name\n";
        }

        if ($this->content) {
            echo "The content of the file is " . $this->content . "\n";
        } else {
            echo "The file doesn't have any content\n";
        }
    }
}